akka-http-tests/src/test/scala/akka/http/javadsl/server/HttpAppSpec.scala

```diff
@@ -22,48 +23,55 @@ import scala.runtime.BoxedUnit
 class HttpAppSpec extends AkkaSpec with RequestBuilding with Eventually {
   import system.dispatcher
 
-  "HttpApp" should {
+  "HttpApp  Java" should {
 
     "start without ActorSystem" in {
-      val (_, host, port) = TestUtils.temporaryServerHostnameAndPort()
 
       val minimal = new MinimalHttpApp()
 
       val server = Future {
-        minimal.startServer(host, port, ServerSettings.create(ConfigFactory.load))
+        minimal.startServer("localhost", 0, ServerSettings.create(ConfigFactory.load))
```

Comment:
As mentioned in #824 there is a potential race condition when probing a port using `temporaryServerHostnameAndPort `.